src/lib/device-types/index.ts

```diff
@@ -214,7 +246,9 @@ function fetchDeviceTypesAndReschedule(): Promise<Dictionary<DeviceTypeInfo>> {
 		updateDeviceTypesCache(promise);
 	}
 
-	return promise;
+	return promise.tap(deviceTypeInfos => {
+		return syncDataModel(deviceTypeInfos, syncSettings.map);
```

Comment:
* Since the updateOrInsert in here runs on each run of `fetchDeviceTypesAndReschedule` would it make sense to check whether the new device type cache did change at all, or (as mentioned above) add some the respective check in the `updateOrInsert`?
* Also, shouldn't we provide the option to consumer projects of the open-balena-api to only run this once (so that the DB is the source of truth of the known device types after the initial sync)? Or we don't yet care for this yet?

Reply:
I guess we can look at extending `updateOrInsert`, I would not only run this once, because in that case instances would not receive new device types without restarting.